src/vmspawn/vmspawn.c

```diff
@@ -2382,18 +2407,36 @@ static int run_virtual_machine(int kvm_device_fd, int vhost_device_fd) {
 
         if (ARCHITECTURE_SUPPORTS_SMBIOS)
                 FOREACH_ARRAY(cred, arg_credentials.credentials, arg_credentials.n_credentials) {
-                        _cleanup_free_ char *cred_data_b64 = NULL;
+                        _cleanup_free_ char *p, *cred_data_b64 = NULL;
                         ssize_t n;
 
                         n = base64mem(cred->data, cred->size, &cred_data_b64);
                         if (n < 0)
                                 return log_oom();
 
+                        p = path_join(smbios_dir, cred->id);
+                        if (!p)
+                                return log_oom();
+
+                        _cleanup_fclose_ FILE *f = NULL;
+                        f = fopen(p, "wxe");
+                        if (!f)
+                                return log_error_errno(errno, "Failed to open smbios credential file %s: %m", p);
```

Comment:
as above, please use `fopen_tmpfile_linkable()` and `flink_tmpfile()` for new code